### ISSUE TITLE
json.Marshal(events.Event) invoke error

### ISSUE THREAD

**Issue by janmisek1** [SparkPost/gosparkpost#115]
`json.Marshal(events.Event) invoke error` : 
`json: error calling MarshalJSON for type events.LatLong: invalid character '%' looking for beginning of value`

The error is caused by `github.com/SparkPost/gosparkpost/events/events.go:257`

The code dos not marshal float32 value correctly:

here is fix of the line:
`return []byte(fmt.Sprintf("%g", *v)), nil`

**Comment by yargevad** [comment]
Hi @janmisek1 - thanks for the report! I'm going to add some tests for this today, and switch the `%v` to a `%f` since the expected representation for lat/long isn't scientific notation.

**Comment by rantav** [comment]
Hi, I'm seeing this as well. 
Subscribe to webhooks and when event comes with geo data I get this error. 
For example: 
```js
[
  {
    "msys": {
      "track_event": {
        "type": "open",
        "subaccount_id": "1",
        "delv_method": "esmtp",
        "customer_id": "99202",
        "message_id": "0002cd1a315a6f6c1816",
        "ip_address": "66.249.93.204",
        "event_id": "30707178003527610",
        "user_agent": "Mozilla/5.0 (Windows NT 5.1; rv:11.0) Gecko Firefox/11.0 (via ggpht.com GoogleImageProxy)",
        "timestamp": "1513167573",
        "rcpt_to": "ran@yodas.com",
        "campaign_id": "login",
        "template_id": "template_48721579019282784",
        "sending_ip": "52.39.51.184",
        "ip_pool": "shared",
        "friendly_from": "hi@yodas.com",
        "rcpt_tags": [],
        "transmission_id": "48721579019282784",
        "routing_domain": "google.com",
        "subject": "Your login link to Yodas",
        "template_version": "0",
        "msg_size": "13072",
        "rcpt_meta": {},
        "msg_from": "msprvs1=17520WcbWnMNe=bounces-99202-1@spmailtechn.com",
        "geo_ip": {
          "country": "EU",
          "region": "",
          "city": "",
          "latitude": 47,
          "longitude": 8
        },
        "raw_rcpt_to": "ran@yodas.com"
      }
    }
  }
]
```


Another (similar) example: 
```js
[
  {
    "msys": {
      "track_event": {
        "delv_method": "esmtp",
        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.94 Safari/537.36",
        "type": "click",
        "event_id": "66735982185650821",
        "accept_language": "en-US,en;q=0.9,he-IL;q=0.8,he;q=0.7",
        "timestamp": "1513167669",
        "message_id": "0002cd1a315a6f6c1816",
        "subaccount_id": "1",
        "customer_id": "99202",
        "target_link_url": "http://localhost:3000/chat?token=cd03144c-d2d6-11e7-8b7e-7e79e20367fb&src=resume_email",
        "ip_address": "84.108.137.251",
        "rcpt_to": "ran@yodas.com",
        "campaign_id": "login",
        "template_id": "template_48721579019282784",
        "sending_ip": "52.39.51.184",
        "ip_pool": "shared",
        "friendly_from": "hi@yodas.com",
        "rcpt_tags": [],
        "transmission_id": "48721579019282784",
        "routing_domain": "google.com",
        "subject": "Your login link to Yodas",
        "template_version": "0",
        "msg_size": "13072",
        "rcpt_meta": {},
        "msg_from": "msprvs1=17520WcbWnMNe=bounces-99202-1@spmailtechn.com",
        "geo_ip": {
          "country": "IL",
          "region": "04",
          "city": "Pardes Hanna",
          "latitude": 32.4741,
          "longitude": 34.9778
        },
        "raw_rcpt_to": "ran@yodas.com"
      }
    }
  }
]
```

**Comment by yargevad** [comment]
fixed in [PR#126](https://github.com/SparkPost/gosparkpost/pull/126)
